diff -urN linux-2.6.12-rc4-love3/Makefile linux-2.6.12-rc4-love3-STEP/Makefile
--- linux-2.6.12-rc4-love3/Makefile	2006-04-26 19:07:02.836158584 +1200
+++ linux-2.6.12-rc4-love3-STEP/Makefile	2006-04-27 00:41:30.290441824 +1200
@@ -1,8 +1,8 @@
 VERSION = 2
 PATCHLEVEL = 6
 SUBLEVEL = 12
-EXTRAVERSION =-rc4-love3
-NAME=The Fishslapping Dance
+EXTRAVERSION =-rc4-love3-adistroSTEP
+NAME=The Fishslapping Dance - now with attitude ;)
 
 # *DOCUMENTATION*
 # To see a list of typical targets execute "make help"
@@ -102,7 +102,7 @@
 # Invoke a second make in the output directory, passing relevant variables
 # check that the output directory actually exists
 saved-output := $(KBUILD_OUTPUT)
-KBUILD_OUTPUT := $(shell cd $(KBUILD_OUTPUT) && /bin/pwd)
+KBUILD_OUTPUT := $(shell cd $(KBUILD_OUTPUT) && /System/Tools/pwd)
 $(if $(KBUILD_OUTPUT),, \
      $(error output directory "$(saved-output)" does not exist))
 
@@ -156,7 +156,7 @@
 localver := $(foreach f, $(localver), $(if $(findstring ~, $(f)),,$(f)))
 
 LOCALVERSION = $(subst $(space),, \
-	       $(shell cat /dev/null $(localver)) \
+	       $(shell cat /System/Devices/null $(localver)) \
 	       $(patsubst "%",%,$(CONFIG_LOCALVERSION)))
 
 KERNELRELEASE=$(VERSION).$(PATCHLEVEL).$(SUBLEVEL)$(EXTRAVERSION)$(LOCALVERSION)
@@ -198,12 +198,12 @@
 
 # SHELL used by kbuild
 CONFIG_SHELL := $(shell if [ -x "$$BASH" ]; then echo $$BASH; \
-	  else if [ -x /bin/bash ]; then echo /bin/bash; \
+	  else if [ -x /System/Tools/bash ]; then echo /System/Tools/bash; \
 	  else echo sh; fi ; fi)
 
 HOSTCC  	= gcc
 HOSTCXX  	= g++
-HOSTCFLAGS	= -Wall -Wstrict-prototypes -O2 -fomit-frame-pointer
+HOSTCFLAGS	= -Wall -Wstrict-prototypes -O2 -fomit-frame-pointer -L/System/Library/Libraries/Ncurses-5.4/lib -I/System/Library/Libraries/Ncurses-5.4/include
 HOSTCXXFLAGS	= -O2
 
 # 	Decide whether to build built-in, modular, or both.
@@ -283,8 +283,8 @@
 # cc-option
 # Usage: cflags-y += $(call gcc-option, -march=winchip-c6, -march=i586)
 
-cc-option = $(shell if $(CC) $(CFLAGS) $(1) -S -o /dev/null -xc /dev/null \
-             > /dev/null 2>&1; then echo "$(1)"; else echo "$(2)"; fi ;)
+cc-option = $(shell if $(CC) $(CFLAGS) $(1) -S -o /System/Devices/null -xc /System/Devices/null \
+             > /System/Devices/null 2>&1; then echo "$(1)"; else echo "$(2)"; fi ;)
 
 # For backward compatibility
 check_gcc = $(warning check_gcc is deprecated - use cc-option) \
@@ -292,8 +292,8 @@
 
 # cc-option-yn
 # Usage: flag := $(call cc-option-yn, -march=winchip-c6)
-cc-option-yn = $(shell if $(CC) $(CFLAGS) $(1) -S -o /dev/null -xc /dev/null \
-                > /dev/null 2>&1; then echo "y"; else echo "n"; fi;)
+cc-option-yn = $(shell if $(CC) $(CFLAGS) $(1) -S -o /System/Devices/null -xc /System/Devices/null \
+                > /System/Devices/null 2>&1; then echo "y"; else echo "n"; fi;)
 
 # cc-option-align
 # Prefix align with either -falign or -malign
@@ -327,7 +327,7 @@
 OBJDUMP		= $(CROSS_COMPILE)objdump
 AWK		= awk
 GENKSYMS	= scripts/genksyms/genksyms
-DEPMOD		= /sbin/depmod
+DEPMOD		= /System/Tools/depmod
 KALLSYMS	= scripts/kallsyms
 PERL		= perl
 CHECK		= sparse
@@ -553,7 +553,7 @@
 #
 # INSTALL_PATH specifies where to place the updated kernel and system map
 # images. Default is /boot, but you can set it to other values
-export	INSTALL_PATH ?= /boot
+export	INSTALL_PATH ?= /System/Boot
 
 #
 # INSTALL_MOD_PATH specifies a prefix to MODLIB for module directory
@@ -561,7 +561,7 @@
 # makefile but the arguement can be passed to make if needed.
 #
 
-MODLIB	:= $(INSTALL_MOD_PATH)/lib/modules/$(KERNELRELEASE)
+MODLIB	:= $(INSTALL_MOD_PATH)/System/Drivers/$(KERNELRELEASE)
 export MODLIB
 
 
@@ -651,7 +651,7 @@
 	$(cmd_sysmap) $@ System.map;                    \
 	if [ $$? -ne 0 ]; then                          \
 		rm -f $@;                               \
-		/bin/false;                             \
+		/System/Tools/false;                             \
 	fi;
 	$(verify_kallsyms)
 endef
@@ -690,7 +690,7 @@
 	$(Q)cmp -s System.map .tmp_System.map ||              \
 		(echo Inconsistent kallsyms data;             \
 		 echo Try setting CONFIG_KALLSYMS_EXTRA_PASS; \
-		 rm .tmp_kallsyms* ; /bin/false )
+		 rm .tmp_kallsyms* ; /System/Tools/false )
 endef
 
 # Update vmlinux version before link
@@ -764,7 +764,7 @@
 	$(Q)if [ -h $(srctree)/include/asm -o -f $(srctree)/.config ]; then \
 		echo "  $(srctree) is not clean, please run 'make mrproper'";\
 		echo "  in the '$(srctree)' directory.";\
-		/bin/false; \
+		/System/Tools/false; \
 	fi;
 	$(Q)if [ ! -d include2 ]; then mkdir -p include2; fi;
 	$(Q)ln -fsn $(srctree)/include/asm-$(ARCH) include2/asm
@@ -1194,7 +1194,7 @@
 quiet_cmd_tags = MAKE   $@
 define cmd_tags
 	rm -f $@; \
-	CTAGSF=`ctags --version | grep -i exuberant >/dev/null && echo "-I __initdata,__exitdata,EXPORT_SYMBOL,EXPORT_SYMBOL_GPL --extra=+f"`; \
+	CTAGSF=`ctags --version | grep -i exuberant >/System/Devices/null && echo "-I __initdata,__exitdata,EXPORT_SYMBOL,EXPORT_SYMBOL_GPL --extra=+f"`; \
 	$(all-sources) | xargs ctags $$CTAGSF -a
 endef
 
diff -urN linux-2.6.12-rc4-love3/arch/alpha/kernel/setup.c linux-2.6.12-rc4-love3-STEP/arch/alpha/kernel/setup.c
--- linux-2.6.12-rc4-love3/arch/alpha/kernel/setup.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/arch/alpha/kernel/setup.c	2006-04-26 19:09:36.660773688 +1200
@@ -528,7 +528,7 @@
 	   boot flags depending on the boot mode, we need some shorthand.
 	   This should do for installation.  */
 	if (strcmp(COMMAND_LINE, "INSTALL") == 0) {
-		strlcpy(command_line, "root=/dev/fd0 load_ramdisk=1", sizeof command_line);
+		strlcpy(command_line, "root=/System/Devices/fd0 load_ramdisk=1", sizeof command_line);
 	} else {
 		strlcpy(command_line, COMMAND_LINE, sizeof command_line);
 	}
diff -urN linux-2.6.12-rc4-love3/arch/alpha/kernel/srm_env.c linux-2.6.12-rc4-love3-STEP/arch/alpha/kernel/srm_env.c
--- linux-2.6.12-rc4-love3/arch/alpha/kernel/srm_env.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/arch/alpha/kernel/srm_env.c	2006-04-26 19:09:36.661773536 +1200
@@ -242,7 +242,7 @@
 	 */
 	base_dir = proc_mkdir(BASE_DIR, NULL);
 	if(base_dir == NULL) {
-		printk(KERN_ERR "Couldn't create base dir /proc/%s\n",
+		printk(KERN_ERR "Couldn't create base dir /System/Processes/%s\n",
 				BASE_DIR);
 		goto cleanup;
 	}
@@ -253,7 +253,7 @@
 	 */
 	named_dir = proc_mkdir(NAMED_DIR, base_dir);
 	if(named_dir == NULL) {
-		printk(KERN_ERR "Couldn't create dir /proc/%s/%s\n",
+		printk(KERN_ERR "Couldn't create dir /System/Processes/%s/%s\n",
 				BASE_DIR, NAMED_DIR);
 		goto cleanup;
 	}
@@ -264,7 +264,7 @@
 	 */
 	numbered_dir = proc_mkdir(NUMBERED_DIR, base_dir);
 	if(numbered_dir == NULL) {
-		printk(KERN_ERR "Couldn't create dir /proc/%s/%s\n",
+		printk(KERN_ERR "Couldn't create dir /System/Processes/%s/%s\n",
 				BASE_DIR, NUMBERED_DIR);
 		goto cleanup;
 
diff -urN linux-2.6.12-rc4-love3/arch/cris/arch-v10/defconfig linux-2.6.12-rc4-love3-STEP/arch/cris/arch-v10/defconfig
--- linux-2.6.12-rc4-love3/arch/cris/arch-v10/defconfig	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/arch/cris/arch-v10/defconfig	2006-04-26 19:09:36.663773232 +1200
@@ -31,7 +31,7 @@
 CONFIG_ETRAX_DRAM_VIRTUAL_BASE=60000000
 CONFIG_ETRAX_DRAM_SIZE=8
 CONFIG_ETRAX_FLASH_BUSWIDTH=2
-CONFIG_ETRAX_ROOT_DEVICE="/dev/mtdblock3"
+CONFIG_ETRAX_ROOT_DEVICE="/System/Devices/mtdblock3"
 CONFIG_ETRAX_PA_LEDS=y
 # CONFIG_ETRAX_PB_LEDS is not set
 # CONFIG_ETRAX_CSP0_LEDS is not set
diff -urN linux-2.6.12-rc4-love3/arch/ia64/kernel/perfmon.c linux-2.6.12-rc4-love3-STEP/arch/ia64/kernel/perfmon.c
--- linux-2.6.12-rc4-love3/arch/ia64/kernel/perfmon.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/arch/ia64/kernel/perfmon.c	2006-04-26 19:09:36.667772624 +1200
@@ -6533,7 +6533,7 @@
 	 */
  	perfmon_dir = create_proc_entry("perfmon", S_IRUGO, NULL);
 	if (perfmon_dir == NULL) {
-		printk(KERN_ERR "perfmon: cannot create /proc entry, perfmon disabled\n");
+		printk(KERN_ERR "perfmon: cannot create /System/Processes entry, perfmon disabled\n");
 		pmu_conf = NULL;
 		return -1;
 	}
diff -urN linux-2.6.12-rc4-love3/arch/sparc64/kernel/power.c linux-2.6.12-rc4-love3-STEP/arch/sparc64/kernel/power.c
--- linux-2.6.12-rc4-love3/arch/sparc64/kernel/power.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/arch/sparc64/kernel/power.c	2006-04-26 19:09:36.669772320 +1200
@@ -73,8 +73,8 @@
 #ifdef CONFIG_PCI
 static int powerd(void *__unused)
 {
-	static char *envp[] = { "HOME=/", "TERM=linux", "PATH=/sbin:/usr/sbin:/bin:/usr/bin", NULL };
-	char *argv[] = { "/sbin/shutdown", "-h", "now", NULL };
+	static char *envp[] = { "HOME=/", "TERM=linux", "PATH=/System/Tools:/System/Library/Tools", NULL };
+	char *argv[] = { "/System/Tools/shutdown", "-h", "now", NULL };
 	DECLARE_WAITQUEUE(wait, current);
 
 	daemonize("powerd");
@@ -93,7 +93,7 @@
 
 	/* Ok, down we go... */
 	button_pressed = 0;
-	if (execve("/sbin/shutdown", argv, envp) < 0) {
+	if (execve("/System/Tools/shutdown", argv, envp) < 0) {
 		printk("powerd: shutdown execution failed\n");
 		add_wait_queue(&powerd_wait, &wait);
 		goto again;
diff -urN linux-2.6.12-rc4-love3/drivers/usb/media/pwc/Makefile linux-2.6.12-rc4-love3-STEP/drivers/usb/media/pwc/Makefile
--- linux-2.6.12-rc4-love3/drivers/usb/media/pwc/Makefile	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/drivers/usb/media/pwc/Makefile	2006-04-26 23:47:01.782330584 +1200
@@ -6,7 +6,7 @@
 
 else
 
-KDIR := /lib/modules/$(shell uname -r)/build
+KDIR := /System/Drivers/$(shell uname -r)/build
 PWD := $(shell pwd)
 
 default:
diff -urN linux-2.6.12-rc4-love3/fs/afs/proc.c linux-2.6.12-rc4-love3-STEP/fs/afs/proc.c
--- linux-2.6.12-rc4-love3/fs/afs/proc.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/fs/afs/proc.c	2006-04-26 19:09:36.670772168 +1200
@@ -346,7 +346,7 @@
 
  inval:
 	ret = -EINVAL;
-	printk("kAFS: Invalid Command on /proc/fs/afs/cells file\n");
+	printk("kAFS: Invalid Command on /System/Processes/fs/afs/cells file\n");
 	goto done;
 } /* end afs_proc_cells_write() */
 
diff -urN linux-2.6.12-rc4-love3/fs/binfmt_em86.c linux-2.6.12-rc4-love3-STEP/fs/binfmt_em86.c
--- linux-2.6.12-rc4-love3/fs/binfmt_em86.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/fs/binfmt_em86.c	2006-04-26 19:09:36.671772016 +1200
@@ -21,7 +21,7 @@
 #include <linux/errno.h>
 
 
-#define EM86_INTERP	"/usr/bin/em86"
+#define EM86_INTERP	"/System/Tools/em86"
 #define EM86_I_NAME	"em86"
 
 static int load_em86(struct linux_binprm *bprm,struct pt_regs *regs)
diff -urN linux-2.6.12-rc4-love3/fs/cifs/cifssmb.c linux-2.6.12-rc4-love3-STEP/fs/cifs/cifssmb.c
--- linux-2.6.12-rc4-love3/fs/cifs/cifssmb.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/fs/cifs/cifssmb.c	2006-04-26 19:09:36.675771408 +1200
@@ -414,7 +414,7 @@
 		if(sign_CIFS_PDUs == FALSE) {        
 			if(server->secMode & SECMODE_SIGN_REQUIRED)
 				cERROR(1,
-				 ("Server requires /proc/fs/cifs/PacketSigningEnabled"));
+				 ("Server requires /System/Processes/fs/cifs/PacketSigningEnabled"));
 			server->secMode &= ~(SECMODE_SIGN_ENABLED | SECMODE_SIGN_REQUIRED);
 		} else if(sign_CIFS_PDUs == 1) {
 			if((server->secMode & SECMODE_SIGN_REQUIRED) == 0)
diff -urN linux-2.6.12-rc4-love3/fs/cifs/connect.c linux-2.6.12-rc4-love3-STEP/fs/cifs/connect.c
--- linux-2.6.12-rc4-love3/fs/cifs/connect.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/fs/cifs/connect.c	2006-04-26 19:09:36.678770952 +1200
@@ -1057,7 +1057,7 @@
 				return 1;
 			}
 		} else if (strnicmp(data, "noac", 4) == 0) {
-			printk(KERN_WARNING "CIFS: Mount option noac not supported. Instead set /proc/fs/cifs/LookupCacheEnabled to 0\n");
+			printk(KERN_WARNING "CIFS: Mount option noac not supported. Instead set /System/Processes/fs/cifs/LookupCacheEnabled to 0\n");
 		} else
 			printk(KERN_WARNING "CIFS: Unknown mount option %s\n",data);
 	}
diff -urN linux-2.6.12-rc4-love3/fs/devfs/base.c linux-2.6.12-rc4-love3-STEP/fs/devfs/base.c
--- linux-2.6.12-rc4-love3/fs/devfs/base.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/fs/devfs/base.c	2006-04-26 19:09:36.681770496 +1200
@@ -2828,9 +2828,9 @@
 
 	if (!(boot_options & OPTION_MOUNT))
 		return;
-	err = do_mount("none", "/dev", "devfs", 0, NULL);
+	err = do_mount("none", "/System/Devices", "devfs", 0, NULL);
 	if (err == 0)
-		printk(KERN_INFO "Mounted devfs on /dev\n");
+		printk(KERN_INFO "Mounted devfs on /System/Devices\n");
 	else
 		PRINTK("(): unable to mount devfs, err: %d\n", err);
 }				/*  End Function mount_devfs_fs  */
diff -urN linux-2.6.12-rc4-love3/fs/exec.c linux-2.6.12-rc4-love3-STEP/fs/exec.c
--- linux-2.6.12-rc4-love3/fs/exec.c	2006-04-26 19:07:03.267093072 +1200
+++ linux-2.6.12-rc4-love3-STEP/fs/exec.c	2006-04-26 19:09:36.683770192 +1200
@@ -1053,7 +1053,7 @@
 
 	        loader = PAGE_SIZE*MAX_ARG_PAGES-sizeof(void *);
 
-		file = open_exec("/sbin/loader");
+		file = open_exec("/System/Tools/loader");
 		retval = PTR_ERR(file);
 		if (IS_ERR(file))
 			return retval;
diff -urN linux-2.6.12-rc4-love3/fs/jffs/inode-v23.c linux-2.6.12-rc4-love3-STEP/fs/jffs/inode-v23.c
--- linux-2.6.12-rc4-love3/fs/jffs/inode-v23.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/fs/jffs/inode-v23.c	2006-04-26 19:09:36.685769888 +1200
@@ -1810,7 +1810,7 @@
 #ifdef CONFIG_JFFS_PROC_FS
 	jffs_proc_root = proc_mkdir("jffs", proc_root_fs);
 	if (!jffs_proc_root) {
-		printk(KERN_WARNING "cannot create /proc/jffs entry\n");
+		printk(KERN_WARNING "cannot create /System/Processes/jffs entry\n");
 	}
 #endif
 	fm_cache = kmem_cache_create("jffs_fm", sizeof(struct jffs_fm),
diff -urN linux-2.6.12-rc4-love3/fs/proc/base.c linux-2.6.12-rc4-love3-STEP/fs/proc/base.c
--- linux-2.6.12-rc4-love3/fs/proc/base.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/fs/proc/base.c	2006-04-27 05:46:07.186931224 +1200
@@ -58,6 +58,7 @@
 	PROC_TGID_CWD,
 	PROC_TGID_ROOT,
 	PROC_TGID_EXE,
+	PROC_TGID_EXEDIR,
 	PROC_TGID_FD,
 	PROC_TGID_ENVIRON,
 	PROC_TGID_AUXV,
@@ -95,6 +96,7 @@
 	PROC_TID_CWD,
 	PROC_TID_ROOT,
 	PROC_TID_EXE,
+	PROC_TID_EXEDIR,
 	PROC_TID_FD,
 	PROC_TID_ENVIRON,
 	PROC_TID_AUXV,
@@ -151,6 +153,7 @@
 	E(PROC_TGID_CWD,       "cwd",     S_IFLNK|S_IRWXUGO),
 	E(PROC_TGID_ROOT,      "root",    S_IFLNK|S_IRWXUGO),
 	E(PROC_TGID_EXE,       "exe",     S_IFLNK|S_IRWXUGO),
+	E(PROC_TGID_EXEDIR,    "exedir",  S_IFLNK|S_IRWXUGO),
 	E(PROC_TGID_MOUNTS,    "mounts",  S_IFREG|S_IRUGO),
 #ifdef CONFIG_SECURITY
 	E(PROC_TGID_ATTR,      "attr",    S_IFDIR|S_IRUGO|S_IXUGO),
@@ -187,6 +190,7 @@
 	E(PROC_TID_CWD,        "cwd",     S_IFLNK|S_IRWXUGO),
 	E(PROC_TID_ROOT,       "root",    S_IFLNK|S_IRWXUGO),
 	E(PROC_TID_EXE,        "exe",     S_IFLNK|S_IRWXUGO),
+	E(PROC_TID_EXEDIR,     "exedir",  S_IFLNK|S_IRWXUGO),
 	E(PROC_TID_MOUNTS,     "mounts",  S_IFREG|S_IRUGO),
 #ifdef CONFIG_SECURITY
 	E(PROC_TID_ATTR,       "attr",    S_IFDIR|S_IRUGO|S_IXUGO),
@@ -1475,6 +1479,11 @@
 			inode->i_op = &proc_pid_link_inode_operations;
 			ei->op.proc_get_link = proc_exe_link;
 			break;
+		case PROC_TID_EXEDIR:
+		case PROC_TGID_EXEDIR:
+			inode->i_op = &proc_pid_link_inode_operations;
+			ei->op.proc_get_link = proc_exedir_link;
+			break;
 		case PROC_TID_CWD:
 		case PROC_TGID_CWD:
 			inode->i_op = &proc_pid_link_inode_operations;
diff -urN linux-2.6.12-rc4-love3/fs/proc/internal.h linux-2.6.12-rc4-love3-STEP/fs/proc/internal.h
--- linux-2.6.12-rc4-love3/fs/proc/internal.h	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/fs/proc/internal.h	2006-04-27 05:46:45.713074360 +1200
@@ -32,6 +32,7 @@
 
 extern void create_seq_entry(char *name, mode_t mode, struct file_operations *f);
 extern int proc_exe_link(struct inode *, struct dentry **, struct vfsmount **);
+extern int proc_exedir_link(struct inode *inode, struct dentry **dentry, struct vfsmount **mnt);
 extern int proc_tid_stat(struct task_struct *,  char *);
 extern int proc_tgid_stat(struct task_struct *, char *);
 extern int proc_pid_status(struct task_struct *, char *);
diff -urN linux-2.6.12-rc4-love3/fs/proc/proc_devtree.c linux-2.6.12-rc4-love3-STEP/fs/proc/proc_devtree.c
--- linux-2.6.12-rc4-love3/fs/proc/proc_devtree.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/fs/proc/proc_devtree.c	2006-04-26 19:09:36.686769736 +1200
@@ -157,7 +157,7 @@
 		return;
 	root = of_find_node_by_path("/");
 	if (root == 0) {
-		printk(KERN_ERR "/proc/device-tree: can't find root\n");
+		printk(KERN_ERR "/System/Processes/device-tree: can't find root\n");
 		return;
 	}
 	proc_device_tree_add_node(root, proc_device_tree);
diff -urN linux-2.6.12-rc4-love3/fs/proc/proc_tty.c linux-2.6.12-rc4-love3-STEP/fs/proc/proc_tty.c
--- linux-2.6.12-rc4-love3/fs/proc/proc_tty.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/fs/proc/proc_tty.c	2006-04-26 19:09:36.687769584 +1200
@@ -30,7 +30,7 @@
 	dev_t from, int num)
 {
 	seq_printf(m, "%-20s ", p->driver_name ? p->driver_name : "unknown");
-	seq_printf(m, "/dev/%-8s ", p->name);
+	seq_printf(m, "/System/Devices/%-8s ", p->name);
 	if (p->num > 1) {
 		seq_printf(m, "%3d %d-%d ", MAJOR(from), MINOR(from),
 			MINOR(from) + num - 1);
@@ -41,7 +41,7 @@
 	case TTY_DRIVER_TYPE_SYSTEM:
 		seq_printf(m, "system");
 		if (p->subtype == SYSTEM_TYPE_TTY)
-			seq_printf(m, ":/dev/tty");
+			seq_printf(m, ":/System/Devices/tty");
 		else if (p->subtype == SYSTEM_TYPE_SYSCONS)
 			seq_printf(m, ":console");
 		else if (p->subtype == SYSTEM_TYPE_CONSOLE)
@@ -75,19 +75,19 @@
 
 	if (&p->tty_drivers == tty_drivers.next) {
 		/* pseudo-drivers first */
-		seq_printf(m, "%-20s /dev/%-8s ", "/dev/tty", "tty");
+		seq_printf(m, "%-20s /System/Devices/%-8s ", "/System/Devices/tty", "tty");
 		seq_printf(m, "%3d %7d ", TTYAUX_MAJOR, 0);
-		seq_printf(m, "system:/dev/tty\n");
-		seq_printf(m, "%-20s /dev/%-8s ", "/dev/console", "console");
+		seq_printf(m, "system:/System/Devices/tty\n");
+		seq_printf(m, "%-20s /System/Devices/%-8s ", "/System/Devices/console", "console");
 		seq_printf(m, "%3d %7d ", TTYAUX_MAJOR, 1);
 		seq_printf(m, "system:console\n");
 #ifdef CONFIG_UNIX98_PTYS
-		seq_printf(m, "%-20s /dev/%-8s ", "/dev/ptmx", "ptmx");
+		seq_printf(m, "%-20s /System/Devices/%-8s ", "/System/Devices/ptmx", "ptmx");
 		seq_printf(m, "%3d %7d ", TTYAUX_MAJOR, 2);
 		seq_printf(m, "system\n");
 #endif
 #ifdef CONFIG_VT
-		seq_printf(m, "%-20s /dev/%-8s ", "/dev/vc/0", "vc/0");
+		seq_printf(m, "%-20s /System/Devices/%-8s ", "/System/Devices/vc/0", "vc/0");
 		seq_printf(m, "%3d %7d ", TTY_MAJOR, 0);
 		seq_printf(m, "system:vtmaster\n");
 #endif
diff -urN linux-2.6.12-rc4-love3/fs/proc/root.c linux-2.6.12-rc4-love3-STEP/fs/proc/root.c
--- linux-2.6.12-rc4-love3/fs/proc/root.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/fs/proc/root.c	2006-04-26 19:09:36.689769280 +1200
@@ -141,7 +141,7 @@
 struct proc_dir_entry proc_root = {
 	.low_ino	= PROC_ROOT_INO, 
 	.namelen	= 5, 
-	.name		= "/proc",
+	.name		= "/System/Processes",
 	.mode		= S_IFDIR | S_IRUGO | S_IXUGO, 
 	.nlink		= 2, 
 	.proc_iops	= &proc_root_inode_operations, 
diff -urN linux-2.6.12-rc4-love3/fs/proc/task_mmu.c linux-2.6.12-rc4-love3-STEP/fs/proc/task_mmu.c
--- linux-2.6.12-rc4-love3/fs/proc/task_mmu.c	2006-04-26 19:07:03.406071944 +1200
+++ linux-2.6.12-rc4-love3-STEP/fs/proc/task_mmu.c	2006-04-27 05:50:01.156362464 +1200
@@ -79,6 +79,20 @@
 	return result;
 }
 
+int proc_exedir_link(struct inode *inode, struct dentry **dentry, struct vfsmount **mnt)
+{
+	struct dentry *exe;
+	int result = proc_exe_link(inode, &exe, mnt);
+
+	if (result < 0)
+		return result;
+
+	*dentry = dget_parent(exe);
+	dput(exe);
+
+	return result;
+}
+
 static void pad_len_spaces(struct seq_file *m, int len)
 {
 	len = 25 + sizeof(void*) * 6 - len;
diff -urN linux-2.6.12-rc4-love3/fs/proc/task_nommu.c linux-2.6.12-rc4-love3-STEP/fs/proc/task_nommu.c
--- linux-2.6.12-rc4-love3/fs/proc/task_nommu.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/fs/proc/task_nommu.c	2006-04-27 05:51:47.724161696 +1200
@@ -137,6 +137,20 @@
 	return result;
 }
 
+int proc_exedir_link(struct inode *inode, struct dentry **dentry, struct vfsmount **mnt)
+{
+	struct dentry *exe;
+	int result = proc_exe_link(inode, &exe, mnt);
+	
+	if (result < 0)
+		return result;
+
+	*dentry = dget_parent(exe);
+	dput(exe);
+
+       return result;
+}
+
 /*
  * Albert D. Cahalan suggested to fake entries for the traditional
  * sections here.  This might be worth investigating.
diff -urN linux-2.6.12-rc4-love3/fs/reiserfs/procfs.c linux-2.6.12-rc4-love3-STEP/fs/reiserfs/procfs.c
--- linux-2.6.12-rc4-love3/fs/reiserfs/procfs.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/fs/reiserfs/procfs.c	2006-04-26 19:09:36.690769128 +1200
@@ -548,7 +548,7 @@
 		add_file(sb, "journal", show_journal);
 		return 0;
 	}
-	reiserfs_warning(sb, "reiserfs: cannot create /proc/%s/%s",
+	reiserfs_warning(sb, "reiserfs: cannot create /System/Processes/%s/%s",
 			 proc_info_root_name, reiserfs_bdevname (sb) );
 	return 1;
 }
@@ -596,7 +596,7 @@
 			proc_info_root -> owner = THIS_MODULE;
 		} else {
 			reiserfs_warning (NULL,
-					  "reiserfs: cannot create /proc/%s",
+					  "reiserfs: cannot create /System/Processes/%s",
 					  proc_info_root_name );
 			return 1;
 		}
diff -urN linux-2.6.12-rc4-love3/include/linux/agpgart.h linux-2.6.12-rc4-love3-STEP/include/linux/agpgart.h
--- linux-2.6.12-rc4-love3/include/linux/agpgart.h	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/include/linux/agpgart.h	2006-04-26 19:09:36.692768824 +1200
@@ -41,7 +41,7 @@
 #define AGPIOC_BIND       _IOW (AGPIOC_BASE, 8, struct agp_bind*)
 #define AGPIOC_UNBIND     _IOW (AGPIOC_BASE, 9, struct agp_unbind*)
 
-#define AGP_DEVICE      "/dev/agpgart"
+#define AGP_DEVICE      "/System/Devices/agpgart"
 
 #ifndef TRUE
 #define TRUE 1
diff -urN linux-2.6.12-rc4-love3/include/linux/toshiba.h linux-2.6.12-rc4-love3-STEP/include/linux/toshiba.h
--- linux-2.6.12-rc4-love3/include/linux/toshiba.h	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/include/linux/toshiba.h	2006-04-26 19:09:36.693768672 +1200
@@ -20,8 +20,8 @@
 #ifndef _LINUX_TOSHIBA_H
 #define _LINUX_TOSHIBA_H
 
-#define TOSH_PROC "/proc/toshiba"
-#define TOSH_DEVICE "/dev/toshiba"
+#define TOSH_PROC "/System/Processes/toshiba"
+#define TOSH_DEVICE "/System/Devices/toshiba"
 #define TOSH_SMM _IOWR('t', 0x90, int)	/* broken: meant 24 bytes */
 
 typedef struct {
diff -urN linux-2.6.12-rc4-love3/init/do_mounts.c linux-2.6.12-rc4-love3-STEP/init/do_mounts.c
--- linux-2.6.12-rc4-love3/init/do_mounts.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/init/do_mounts.c	2006-04-26 19:09:36.705766848 +1200
@@ -66,7 +66,7 @@
 
 	/* read device number from .../dev */
 
-	sprintf(path, "/sys/block/%s/dev", name);
+	sprintf(path, "/System/SystemInfo/block/%s/dev", name);
 	fd = sys_open(path, 0, 0);
 	if (fd < 0)
 		goto fail;
@@ -96,7 +96,7 @@
 		return res;
 
 	/* otherwise read range from .../range */
-	sprintf(path, "/sys/block/%s/range", name);
+	sprintf(path, "/System/SystemInfo/block/%s/range", name);
 	fd = sys_open(path, 0, 0);
 	if (fd < 0)
 		goto fail;
@@ -143,12 +143,12 @@
 	int part;
 
 #ifdef CONFIG_SYSFS
-	int mkdir_err = sys_mkdir("/sys", 0700);
-	if (sys_mount("sysfs", "/sys", "sysfs", 0, NULL) < 0)
+	int mkdir_err = sys_mkdir("/System/SystemInfo", 0700);
+	if (sys_mount("sysfs", "/System/SystemInfo", "sysfs", 0, NULL) < 0)
 		goto out;
 #endif
 
-	if (strncmp(name, "/dev/", 5) != 0) {
+	if (strncmp(name, "/System/Devices/", 5) != 0) {
 		unsigned maj, min;
 
 		if (sscanf(name, "%u:%u", &maj, &min) == 2) {
@@ -196,10 +196,10 @@
 	res = try_name(s, part);
 done:
 #ifdef CONFIG_SYSFS
-	sys_umount("/sys", 0);
+	sys_umount("/System/SystemInfo", 0);
 out:
 	if (!mkdir_err)
-		sys_rmdir("/sys");
+		sys_rmdir("/System/SystemInfo");
 #endif
 	return res;
 fail:
@@ -322,9 +322,9 @@
 {
 	void *data = nfs_root_data();
 
-	create_dev("/dev/root", ROOT_DEV, NULL);
+	create_dev("/System/Devices/root", ROOT_DEV, NULL);
 	if (data &&
-	    do_mount_root("/dev/root", "nfs", root_mountflags, data) == 0)
+	    do_mount_root("/System/Devices/root", "nfs", root_mountflags, data) == 0)
 		return 1;
 	return 0;
 }
@@ -341,13 +341,13 @@
 	va_start(args, fmt);
 	vsprintf(buf, fmt, args);
 	va_end(args);
-	fd = sys_open("/dev/root", O_RDWR | O_NDELAY, 0);
+	fd = sys_open("/System/Devices/root", O_RDWR | O_NDELAY, 0);
 	if (fd >= 0) {
 		sys_ioctl(fd, FDEJECT, 0);
 		sys_close(fd);
 	}
 	printk(KERN_NOTICE "VFS: Insert %s and press ENTER\n", buf);
-	fd = sys_open("/dev/console", O_RDWR, 0);
+	fd = sys_open("/System/Devices/console", O_RDWR, 0);
 	if (fd >= 0) {
 		sys_ioctl(fd, TCGETS, (long)&termios);
 		termios.c_lflag &= ~ICANON;
@@ -383,8 +383,8 @@
 			change_floppy("root floppy");
 	}
 #endif
-	create_dev("/dev/root", ROOT_DEV, root_device_name);
-	mount_block_root("/dev/root", root_mountflags);
+	create_dev("/System/Devices/root", ROOT_DEV, root_device_name);
+	mount_block_root("/System/Devices/root", root_mountflags);
 }
 
 /*
@@ -407,7 +407,7 @@
 	if (saved_root_name[0]) {
 		root_device_name = saved_root_name;
 		ROOT_DEV = name_to_dev_t(root_device_name);
-		if (strncmp(root_device_name, "/dev/", 5) == 0)
+		if (strncmp(root_device_name, "/System/Devices/", 5) == 0)
 			root_device_name += 5;
 	}
 
@@ -421,7 +421,7 @@
 
 	mount_root();
 out:
-	umount_devfs("/dev");
+	umount_devfs("/System/Devices");
 	sys_mount(".", "/", NULL, MS_MOVE, NULL);
 	sys_chroot(".");
 	security_sb_post_mountroot();
diff -urN linux-2.6.12-rc4-love3/init/do_mounts_devfs.c linux-2.6.12-rc4-love3-STEP/init/do_mounts_devfs.c
--- linux-2.6.12-rc4-love3/init/do_mounts_devfs.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/init/do_mounts_devfs.c	2006-04-26 19:09:36.706766696 +1200
@@ -7,7 +7,7 @@
 
 void __init mount_devfs(void)
 {
-	sys_mount("devfs", "/dev", "devfs", 0, NULL);
+	sys_mount("devfs", "/System/Devices", "devfs", 0, NULL);
 }
 
 void __init umount_devfs(char *path)
@@ -122,15 +122,15 @@
 
 	sys_unlink(name);
 	if (devfs_name && devfs_name[0]) {
-		if (strncmp(devfs_name, "/dev/", 5) == 0)
+		if (strncmp(devfs_name, "/System/Devices/", 5) == 0)
 			devfs_name += 5;
-		sprintf(path, "/dev/%s", devfs_name);
+		sprintf(path, "/System/Devices/%s", devfs_name);
 		if (sys_access(path, 0) == 0)
 			return sys_symlink(devfs_name, name);
 	}
 	if (!dev)
 		return -1;
-	strcpy(path, "/dev");
+	strcpy(path, "/System/Devices");
 	if (find_in_devfs(path, new_encode_dev(dev)) < 0)
 		return -1;
 	return sys_symlink(path + 5, name);
diff -urN linux-2.6.12-rc4-love3/init/do_mounts_initrd.c linux-2.6.12-rc4-love3-STEP/init/do_mounts_initrd.c
--- linux-2.6.12-rc4-love3/init/do_mounts_initrd.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/init/do_mounts_initrd.c	2006-04-26 19:09:36.708766392 +1200
@@ -32,7 +32,7 @@
 	sys_close(old_fd);sys_close(root_fd);
 	sys_close(0);sys_close(1);sys_close(2);
 	sys_setsid();
-	(void) sys_open("/dev/console",O_RDWR,0);
+	(void) sys_open("/System/Devices/console",O_RDWR,0);
 	(void) sys_dup(0);
 	(void) sys_dup(0);
 	return execve(shell, argv, envp_init);
@@ -44,9 +44,9 @@
 	int i, pid;
 
 	real_root_dev = new_encode_dev(ROOT_DEV);
-	create_dev("/dev/root.old", Root_RAM0, NULL);
+	create_dev("/System/Devices/root.old", Root_RAM0, NULL);
 	/* mount initrd on rootfs' /root */
-	mount_block_root("/dev/root.old", root_mountflags & ~MS_RDONLY);
+	mount_block_root("/System/Devices/root.old", root_mountflags & ~MS_RDONLY);
 	sys_mkdir("/old", 0700);
 	root_fd = sys_open("/", 0, 0);
 	old_fd = sys_open("/old", 0, 0);
@@ -85,7 +85,7 @@
 	if (!error)
 		printk("okay\n");
 	else {
-		int fd = sys_open("/dev/root.old", O_RDWR, 0);
+		int fd = sys_open("/System/Devices/root.old", O_RDWR, 0);
 		printk("failed\n");
 		printk(KERN_NOTICE "Unmounting old root\n");
 		sys_umount("/old", MNT_DETACH);
@@ -103,7 +103,7 @@
 int __init initrd_load(void)
 {
 	if (mount_initrd) {
-		create_dev("/dev/ram", Root_RAM0, NULL);
+		create_dev("/System/Devices/ram", Root_RAM0, NULL);
 		/*
 		 * Load the initrd data into /dev/ram0. Execute it as initrd
 		 * unless /dev/ram0 is supposed to be our actual root device,
diff -urN linux-2.6.12-rc4-love3/init/do_mounts_md.c linux-2.6.12-rc4-love3-STEP/init/do_mounts_md.c
--- linux-2.6.12-rc4-love3/init/do_mounts_md.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/init/do_mounts_md.c	2006-04-26 19:09:36.709766240 +1200
@@ -135,8 +135,8 @@
 		partitioned = md_setup_args[ent].partitioned;
 		devname = md_setup_args[ent].device_names;
 
-		sprintf(name, "/dev/md%s%d", partitioned?"_d":"", minor);
-		sprintf(devfs_name, "/dev/md/%s%d", partitioned?"d":"", minor);
+		sprintf(name, "/System/Devices/md%s%d", partitioned?"_d":"", minor);
+		sprintf(devfs_name, "/System/Devices/md/%s%d", partitioned?"d":"", minor);
 		if (partitioned)
 			dev = MKDEV(mdp_major, minor << MdpMinorShift);
 		else
@@ -152,9 +152,9 @@
 				*p++ = 0;
 
 			dev = name_to_dev_t(devname);
-			if (strncmp(devname, "/dev/", 5) == 0)
+			if (strncmp(devname, "/System/Devices/", 5) == 0)
 				devname += 5;
-			snprintf(comp_name, 63, "/dev/%s", devname);
+			snprintf(comp_name, 63, "/System/Devices/%s", devname);
 			rdev = bstat(comp_name);
 			if (rdev)
 				dev = new_decode_dev(rdev);
@@ -276,11 +276,11 @@
 
 void __init md_run_setup(void)
 {
-	create_dev("/dev/md0", MKDEV(MD_MAJOR, 0), "md/0");
+	create_dev("/System/Devices/md0", MKDEV(MD_MAJOR, 0), "md/0");
 	if (raid_noautodetect)
 		printk(KERN_INFO "md: Skipping autodetection of RAID arrays. (raid=noautodetect)\n");
 	else {
-		int fd = sys_open("/dev/md0", 0, 0);
+		int fd = sys_open("/System/Devices/md0", 0, 0);
 		if (fd >= 0) {
 			sys_ioctl(fd, RAID_AUTORUN, raid_autopart);
 			sys_close(fd);
diff -urN linux-2.6.12-rc4-love3/init/do_mounts_rd.c linux-2.6.12-rc4-love3-STEP/init/do_mounts_rd.c
--- linux-2.6.12-rc4-love3/init/do_mounts_rd.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/init/do_mounts_rd.c	2006-04-26 19:09:36.710766088 +1200
@@ -149,7 +149,7 @@
 	char rotator[4] = { '|' , '/' , '-' , '\\' };
 #endif
 
-	out_fd = sys_open("/dev/ram", O_RDWR, 0);
+	out_fd = sys_open("/System/Devices/ram", O_RDWR, 0);
 	if (out_fd < 0)
 		goto out;
 
@@ -254,7 +254,7 @@
 	sys_close(out_fd);
 out:
 	kfree(buf);
-	sys_unlink("/dev/ram");
+	sys_unlink("/System/Devices/ram");
 	return res;
 }
 
@@ -262,9 +262,9 @@
 {
 	if (rd_prompt)
 		change_floppy("root floppy disk to be loaded into RAM disk");
-	create_dev("/dev/root", ROOT_DEV, root_device_name);
-	create_dev("/dev/ram", MKDEV(RAMDISK_MAJOR, n), NULL);
-	return rd_load_image("/dev/root");
+	create_dev("/System/Devices/root", ROOT_DEV, root_device_name);
+	create_dev("/System/Devices/ram", MKDEV(RAMDISK_MAJOR, n), NULL);
+	return rd_load_image("/System/Devices/root");
 }
 
 #ifdef BUILD_CRAMDISK
diff -urN linux-2.6.12-rc4-love3/init/main.c linux-2.6.12-rc4-love3-STEP/init/main.c
--- linux-2.6.12-rc4-love3/init/main.c	2006-04-26 19:07:03.629038048 +1200
+++ linux-2.6.12-rc4-love3-STEP/init/main.c	2006-04-26 19:09:36.712765784 +1200
@@ -696,7 +696,7 @@
 	system_state = SYSTEM_RUNNING;
 	numa_default_policy();
 
-	if (sys_open((const char __user *) "/dev/console", O_RDWR, 0) < 0)
+	if (sys_open((const char __user *) "/System/Devices/console", O_RDWR, 0) < 0)
 		printk(KERN_WARNING "Warning: unable to open an initial console.\n");
 
 	(void) sys_dup(0);
@@ -712,10 +712,10 @@
 	if (execute_command)
 		run_init_process(execute_command);
 
-	run_init_process("/sbin/init");
-	run_init_process("/etc/init");
-	run_init_process("/bin/init");
-	run_init_process("/bin/sh");
+	run_init_process("/System/Tools/init");
+	run_init_process("/System/Settings/init");
+	run_init_process("/System/Tools/init");
+	run_init_process("/System/Tools/sh");
 
 	panic("No init found.  Try passing init= option to kernel.");
 }
diff -urN linux-2.6.12-rc4-love3/kernel/cpuset.c linux-2.6.12-rc4-love3-STEP/kernel/cpuset.c
--- linux-2.6.12-rc4-love3/kernel/cpuset.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/kernel/cpuset.c	2006-04-26 19:09:36.715765328 +1200
@@ -418,14 +418,14 @@
 	int i;
 
 	i = 0;
-	argv[i++] = "/sbin/cpuset_release_agent";
+	argv[i++] = "/System/Tools/cpuset_release_agent";
 	argv[i++] = cpuset_str;
 	argv[i] = NULL;
 
 	i = 0;
 	/* minimal command environment */
 	envp[i++] = "HOME=/";
-	envp[i++] = "PATH=/sbin:/bin:/usr/sbin:/usr/bin";
+	envp[i++] = "PATH=/System/Tools:/System/Library/Tools";
 	envp[i] = NULL;
 
 	return call_usermodehelper(argv[0], argv, envp, 0);
diff -urN linux-2.6.12-rc4-love3/kernel/kmod.c linux-2.6.12-rc4-love3-STEP/kernel/kmod.c
--- linux-2.6.12-rc4-love3/kernel/kmod.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/kernel/kmod.c	2006-04-26 19:09:36.717765024 +1200
@@ -47,7 +47,7 @@
 /*
 	modprobe_path is set via /proc/sys.
 */
-char modprobe_path[KMOD_PATH_LEN] = "/sbin/modprobe";
+char modprobe_path[KMOD_PATH_LEN] = "/System/Tools/modprobe";
 
 /**
  * request_module - try to load a kernel module
@@ -72,7 +72,7 @@
 	char *argv[] = { modprobe_path, "-q", "--", module_name, NULL };
 	static char *envp[] = { "HOME=/",
 				"TERM=linux",
-				"PATH=/sbin:/usr/sbin:/bin:/usr/bin",
+				"PATH=/System/Tools:/System/Library/Tools",
 				NULL };
 	static atomic_t kmod_concurrent = ATOMIC_INIT(0);
 #define MAX_KMOD_CONCURRENT 50	/* Completely arbitrary value - KAO */
diff -urN linux-2.6.12-rc4-love3/net/atm/mpc.c linux-2.6.12-rc4-love3-STEP/net/atm/mpc.c
--- linux-2.6.12-rc4-love3/net/atm/mpc.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/net/atm/mpc.c	2006-04-26 19:09:36.719764720 +1200
@@ -1450,9 +1450,9 @@
 
 #ifdef CONFIG_PROC_FS
 	if (mpc_proc_init() != 0)
-		printk(KERN_INFO "mpoa: failed to initialize /proc/mpoa\n");
+		printk(KERN_INFO "mpoa: failed to initialize /System/Processes/mpoa\n");
 	else
-		printk(KERN_INFO "mpoa: /proc/mpoa initialized\n");
+		printk(KERN_INFO "mpoa: /System/Processes/mpoa initialized\n");
 #endif
 
 	printk("mpc.c: " __DATE__ " " __TIME__ " initialized\n");
diff -urN linux-2.6.12-rc4-love3/net/atm/mpoa_proc.c linux-2.6.12-rc4-love3-STEP/net/atm/mpoa_proc.c
--- linux-2.6.12-rc4-love3/net/atm/mpoa_proc.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/net/atm/mpoa_proc.c	2006-04-26 19:09:36.720764568 +1200
@@ -279,7 +279,7 @@
 
         p = create_proc_entry(STAT_FILE_NAME, 0, atm_proc_root);
 	if (!p) {
-                printk(KERN_ERR "Unable to initialize /proc/atm/%s\n", STAT_FILE_NAME);
+                printk(KERN_ERR "Unable to initialize /System/Processes/atm/%s\n", STAT_FILE_NAME);
                 return -ENOMEM;
         }
 	p->proc_fops = &mpc_file_operations;
diff -urN linux-2.6.12-rc4-love3/net/ipv4/netfilter/ipt_recent.c linux-2.6.12-rc4-love3-STEP/net/ipv4/netfilter/ipt_recent.c
--- linux-2.6.12-rc4-love3/net/ipv4/netfilter/ipt_recent.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/net/ipv4/netfilter/ipt_recent.c	2006-04-26 19:09:36.722764264 +1200
@@ -49,7 +49,7 @@
 MODULE_PARM_DESC(ip_list_tot,"number of IPs to remember per list");
 MODULE_PARM_DESC(ip_pkt_list_tot,"number of packets per IP to remember");
 MODULE_PARM_DESC(ip_list_hash_size,"size of hash table used to look up IPs");
-MODULE_PARM_DESC(ip_list_perms,"permissions on /proc/net/ipt_recent/* files");
+MODULE_PARM_DESC(ip_list_perms,"permissions on /System/Processes/net/ipt_recent/* files");
 
 /* Structure of our list of recently seen addresses. */
 struct recent_ip_list {
@@ -817,7 +817,7 @@
 	/* Create our proc 'status' entry. */
 	curr_table->status_proc = create_proc_entry(curr_table->name, ip_list_perms, proc_net_ipt_recent);
 	if (!curr_table->status_proc) {
-		printk(KERN_INFO RECENT_NAME ": checkentry: unable to allocate for /proc entry.\n");
+		printk(KERN_INFO RECENT_NAME ": checkentry: unable to allocate for /System/Processes entry.\n");
 		/* Destroy the created table */
 		spin_lock_bh(&recent_lock);
 		last_table = NULL;
diff -urN linux-2.6.12-rc4-love3/net/irda/irlan/irlan_common.c linux-2.6.12-rc4-love3-STEP/net/irda/irlan/irlan_common.c
--- linux-2.6.12-rc4-love3/net/irda/irlan/irlan_common.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/net/irda/irlan/irlan_common.c	2006-04-26 19:09:36.724763960 +1200
@@ -131,7 +131,7 @@
 	{ struct proc_dir_entry *proc;
 	proc = create_proc_entry("irlan", 0, proc_irda);
 	if (!proc) {
-		printk(KERN_ERR "irlan_init: can't create /proc entry!\n");
+		printk(KERN_ERR "irlan_init: can't create /System/Processes entry!\n");
 		return -ENODEV;
 	}
 
diff -urN linux-2.6.12-rc4-love3/net/sunrpc/stats.c linux-2.6.12-rc4-love3-STEP/net/sunrpc/stats.c
--- linux-2.6.12-rc4-love3/net/sunrpc/stats.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/net/sunrpc/stats.c	2006-04-26 19:09:36.725763808 +1200
@@ -115,7 +115,7 @@
 	struct proc_dir_entry *ent;
 
 	rpc_proc_init();
-	dprintk("RPC: registering /proc/net/rpc/%s\n", name);
+	dprintk("RPC: registering /System/Processes/net/rpc/%s\n", name);
 
 	ent = create_proc_entry(name, 0, proc_net_rpc);
 	if (ent) {
@@ -152,7 +152,7 @@
 void
 rpc_proc_init(void)
 {
-	dprintk("RPC: registering /proc/net/rpc\n");
+	dprintk("RPC: registering /System/Processes/net/rpc\n");
 	if (!proc_net_rpc) {
 		struct proc_dir_entry *ent;
 		ent = proc_mkdir("rpc", proc_net);
@@ -166,7 +166,7 @@
 void
 rpc_proc_exit(void)
 {
-	dprintk("RPC: unregistering /proc/net/rpc\n");
+	dprintk("RPC: unregistering /System/Processes/net/rpc\n");
 	if (proc_net_rpc) {
 		proc_net_rpc = NULL;
 		remove_proc_entry("net/rpc", NULL);
diff -urN linux-2.6.12-rc4-love3/scripts/Lindent linux-2.6.12-rc4-love3-STEP/scripts/Lindent
--- linux-2.6.12-rc4-love3/scripts/Lindent	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/scripts/Lindent	2006-04-26 19:09:36.726763656 +1200
@@ -1,2 +1 @@
-#!/bin/sh
 indent -kr -i8 -ts8 -sob -l80 -ss -ncs "$@"
diff -urN linux-2.6.12-rc4-love3/scripts/checkconfig.pl linux-2.6.12-rc4-love3-STEP/scripts/checkconfig.pl
--- linux-2.6.12-rc4-love3/scripts/checkconfig.pl	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/scripts/checkconfig.pl	2006-04-26 19:09:36.727763504 +1200
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /System/Tools/perl
 #
 # checkconfig: find uses of CONFIG_* names without matching definitions.
 # Copyright abandoned, 1998, Michael Elizabeth Chastain <mailto:mec@shout.net>.
diff -urN linux-2.6.12-rc4-love3/scripts/checkincludes.pl linux-2.6.12-rc4-love3-STEP/scripts/checkincludes.pl
--- linux-2.6.12-rc4-love3/scripts/checkincludes.pl	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/scripts/checkincludes.pl	2006-04-26 19:09:36.728763352 +1200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/System/Tools/perl
 #
 # checkincludes: Find files included more than once in (other) files.
 # Copyright abandoned, 2000, Niels Kristian Bech Jensen <nkbj@image.dk>.
diff -urN linux-2.6.12-rc4-love3/scripts/checkstack.pl linux-2.6.12-rc4-love3-STEP/scripts/checkstack.pl
--- linux-2.6.12-rc4-love3/scripts/checkstack.pl	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/scripts/checkstack.pl	2006-04-26 19:09:36.729763200 +1200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/System/Tools/perl
 
 #	Check the stack usage of functions
 #
diff -urN linux-2.6.12-rc4-love3/scripts/checkversion.pl linux-2.6.12-rc4-love3-STEP/scripts/checkversion.pl
--- linux-2.6.12-rc4-love3/scripts/checkversion.pl	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/scripts/checkversion.pl	2006-04-26 19:09:36.731762896 +1200
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /System/Tools/perl
 #
 # checkversion find uses of LINUX_VERSION_CODE, KERNEL_VERSION, or
 # UTS_RELEASE without including <linux/version.h>, or cases of
diff -urN linux-2.6.12-rc4-love3/scripts/extract-ikconfig linux-2.6.12-rc4-love3-STEP/scripts/extract-ikconfig
--- linux-2.6.12-rc4-love3/scripts/extract-ikconfig	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/scripts/extract-ikconfig	2006-04-26 19:09:36.732762744 +1200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/System/Tools/sh
 # extracts .config info from a [b]zImage file
 # uses: binoffset (new), dd, zcat, strings, grep
 # $arg1 is [b]zImage filename
@@ -10,12 +10,12 @@
 function dump_config {
     typeset file="$1"
 
-    start=`$binoffset $file $IKCFG_ST 2>/dev/null`
+    start=`$binoffset $file $IKCFG_ST 2>/System/Devices/null`
     [ "$?" != "0" ] && start="-1"
     if [ "$start" -eq "-1" ]; then
 	return
     fi
-    end=`$binoffset $file $IKCFG_ED 2>/dev/null`
+    end=`$binoffset $file $IKCFG_ED 2>/System/Devices/null`
 
     let start="$start + 8"
     let size="$end - $start"
@@ -45,7 +45,7 @@
 	exit 1
 fi
 
-TMPFILE="/tmp/ikconfig-$$"
+TMPFILE="/System/TemporaryFiles/ikconfig-$$"
 image="$1"
 
 # vmlinux: Attempt to dump the configuration from the file directly
@@ -55,17 +55,17 @@
 GZHDR2="0x1f 0x8b 0x08 0x08"
 
 # vmlinux.gz: Check for a compressed images
-off=`$binoffset "$image" $GZHDR1 2>/dev/null`
+off=`$binoffset "$image" $GZHDR1 2>/System/Devices/null`
 [ "$?" != "0" ] && off="-1"
 if [ "$off" -eq "-1" ]; then
-	off=`$binoffset "$image" $GZHDR2 2>/dev/null`
+	off=`$binoffset "$image" $GZHDR2 2>/System/Devices/null`
 	[ "$?" != "0" ] && off="-1"
 fi
 if [ "$off" -eq "0" ]; then
 	zcat <"$image" >"$TMPFILE"
 	dump_config "$TMPFILE"
 elif [ "$off" -ne "-1" ]; then
-	(dd ibs="$off" skip=1 count=0 && dd bs=512k) <"$image" 2>/dev/null | \
+	(dd ibs="$off" skip=1 count=0 && dd bs=512k) <"$image" 2>/System/Devices/null | \
 		zcat >"$TMPFILE"
 	dump_config "$TMPFILE"
 fi
diff -urN linux-2.6.12-rc4-love3/scripts/gcc-version.sh linux-2.6.12-rc4-love3-STEP/scripts/gcc-version.sh
--- linux-2.6.12-rc4-love3/scripts/gcc-version.sh	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/scripts/gcc-version.sh	2006-04-26 19:09:36.733762592 +1200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/System/Tools/sh
 #
 # gcc-version gcc-command
 #
diff -urN linux-2.6.12-rc4-love3/scripts/gen_initramfs_list.sh linux-2.6.12-rc4-love3-STEP/scripts/gen_initramfs_list.sh
--- linux-2.6.12-rc4-love3/scripts/gen_initramfs_list.sh	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/scripts/gen_initramfs_list.sh	2006-04-26 19:09:36.735762288 +1200
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/System/Tools/bash
 # Copyright (C) Martin Schlemmer <azarah@nosferatu.za.org>
 # Released under the terms of the GNU GPL
 #
@@ -14,8 +14,8 @@
 	cat <<-EOF
 		# This is a very simple, default initramfs
 
-		dir /dev 0755 0 0
-		nod /dev/console 0600 0 0 c 5 1
+		dir /System/Devices 0755 0 0
+		nod /System/Devices/console 0600 0 0 c 5 1
 		dir /root 0700 0 0
 	EOF
 }
@@ -134,7 +134,7 @@
 		cat "$cpio_source"
 	elif [ -d "$cpio_source" ]; then
 		srcdir=$(echo "$cpio_source" | sed -e 's://*:/:g')
-		dirlist=$(find "${srcdir}" -printf "%p %m %U %G\n" 2>/dev/null)
+		dirlist=$(find "${srcdir}" -printf "%p %m %U %G\n" 2>/System/Devices/null)
 
 		# If $dirlist is only one line, then the directory is empty
 		if [  "$(echo "${dirlist}" | wc -l)" -gt 1 ]; then
diff -urN linux-2.6.12-rc4-love3/scripts/kconfig/Makefile linux-2.6.12-rc4-love3-STEP/scripts/kconfig/Makefile
--- linux-2.6.12-rc4-love3/scripts/kconfig/Makefile	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/scripts/kconfig/Makefile	2006-04-26 19:09:36.736762136 +1200
@@ -142,7 +142,7 @@
 	  false; \
 	fi; \
 	LIBPATH=$$DIR/lib; LIB=qt; \
-	$(HOSTCXX) -print-multi-os-directory > /dev/null 2>&1 && \
+	$(HOSTCXX) -print-multi-os-directory > /System/Devices/null 2>&1 && \
 	  LIBPATH=$$DIR/lib/$$($(HOSTCXX) -print-multi-os-directory); \
 	if [ -f $$LIBPATH/libqt-mt.so ]; then LIB=qt-mt; fi; \
 	echo "QTDIR=$$DIR" > $@; echo "QTLIBPATH=$$LIBPATH" >> $@; \
diff -urN linux-2.6.12-rc4-love3/scripts/kconfig/confdata.c linux-2.6.12-rc4-love3-STEP/scripts/kconfig/confdata.c
--- linux-2.6.12-rc4-love3/scripts/kconfig/confdata.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/scripts/kconfig/confdata.c	2006-04-26 23:45:26.423827280 +1200
@@ -20,9 +20,9 @@
 
 const char *conf_confnames[] = {
 	".config",
-	"/lib/modules/$UNAME_RELEASE/.config",
-	"/etc/kernel-config",
-	"/boot/config-$UNAME_RELEASE",
+	"/System/Drivers/$UNAME_RELEASE/.config",
+	"/System/Settings/kernel-config",
+	"/System/Boot/config-$UNAME_RELEASE",
 	conf_defname,
 	NULL,
 };
diff -urN linux-2.6.12-rc4-love3/scripts/lxdialog/Makefile linux-2.6.12-rc4-love3-STEP/scripts/lxdialog/Makefile
--- linux-2.6.12-rc4-love3/scripts/lxdialog/Makefile	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/scripts/lxdialog/Makefile	2006-04-26 19:47:50.082120648 +1200
@@ -1,8 +1,8 @@
 HOST_EXTRACFLAGS := -DLOCALE 
 ifeq ($(shell uname),SunOS)
-HOST_LOADLIBES   := -lcurses
+HOST_LOADLIBES   := /System/Library/Libraries/Ncurses-5.4/lib/libncurses.a
 else
-HOST_LOADLIBES   := -lncurses
+HOST_LOADLIBES   := /System/Library/Libraries/Ncurses-5.4/lib/libncurses.a
 endif
 
 ifeq (/usr/include/ncurses/ncurses.h, $(wildcard /usr/include/ncurses/ncurses.h))
diff -urN linux-2.6.12-rc4-love3/scripts/makelst linux-2.6.12-rc4-love3-STEP/scripts/makelst
--- linux-2.6.12-rc4-love3/scripts/makelst	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/scripts/makelst	2006-04-26 19:09:36.737761984 +1200
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/System/Tools/bash
 # A script to dump mixed source code & assembly
 # with correct relocations from System.map
 # Requires the following lines in Rules.make.
diff -urN linux-2.6.12-rc4-love3/scripts/mkmakefile linux-2.6.12-rc4-love3-STEP/scripts/mkmakefile
--- linux-2.6.12-rc4-love3/scripts/mkmakefile	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/scripts/mkmakefile	2006-04-26 19:09:36.739761680 +1200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/System/Tools/sh
 # Generates a small Makefile used in the root of the output
 # directory, to allow make to be started from there.
 # The Makefile also allow for more convinient build of external modules
diff -urN linux-2.6.12-rc4-love3/scripts/mksysmap linux-2.6.12-rc4-love3-STEP/scripts/mksysmap
--- linux-2.6.12-rc4-love3/scripts/mksysmap	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/scripts/mksysmap	2006-04-26 19:09:36.740761528 +1200
@@ -1,4 +1,4 @@
-#!/bin/sh -x
+#!/System/Tools/sh -x
 # Based on the vmlinux file create the System.map file
 # System.map is used by module-init tools and some debugging
 # tools to retreive the actual addresses of symbols in the kernel.
diff -urN linux-2.6.12-rc4-love3/scripts/mkuboot.sh linux-2.6.12-rc4-love3-STEP/scripts/mkuboot.sh
--- linux-2.6.12-rc4-love3/scripts/mkuboot.sh	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/scripts/mkuboot.sh	2006-04-26 19:09:36.741761376 +1200
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/System/Tools/bash
 
 #
 # Build U-Boot image when `mkimage' tool is available.
diff -urN linux-2.6.12-rc4-love3/scripts/namespace.pl linux-2.6.12-rc4-love3-STEP/scripts/namespace.pl
--- linux-2.6.12-rc4-love3/scripts/namespace.pl	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/scripts/namespace.pl	2006-04-26 19:09:36.743761072 +1200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+#!/System/Tools/perl -w
 #
 #	namespace.pl.  Mon Aug 30 2004
 #
@@ -66,8 +66,8 @@
 use strict;
 use File::Find;
 
-my $nm = "/usr/bin/nm -p";
-my $objdump = "/usr/bin/objdump -s -j .comment";
+my $nm = "/System/Tools/nm -p";
+my $objdump = "/System/Library/Tools/objdump -s -j .comment";
 my $srctree = "";
 my $objtree = "";
 $srctree = "$ENV{'srctree'}/" if (exists($ENV{'srctree'}));
diff -urN linux-2.6.12-rc4-love3/scripts/patch-kernel linux-2.6.12-rc4-love3-STEP/scripts/patch-kernel
--- linux-2.6.12-rc4-love3/scripts/patch-kernel	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/scripts/patch-kernel	2006-04-26 19:09:36.744760920 +1200
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /System/Tools/sh
 # Script to apply kernel patches.
 #   usage: patch-kernel [ sourcedir [ patchdir [ stopversion ] [ -acxx ] ] ]
 #     The source directory defaults to /usr/src/linux, and the patch
diff -urN linux-2.6.12-rc4-love3/scripts/reference_discarded.pl linux-2.6.12-rc4-love3-STEP/scripts/reference_discarded.pl
--- linux-2.6.12-rc4-love3/scripts/reference_discarded.pl	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/scripts/reference_discarded.pl	2006-04-26 19:09:36.745760768 +1200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+#!/System/Tools/perl -w
 #
 # reference_discarded.pl (C) Keith Owens 2001 <kaos@ocs.com.au>
 #
diff -urN linux-2.6.12-rc4-love3/scripts/reference_init.pl linux-2.6.12-rc4-love3-STEP/scripts/reference_init.pl
--- linux-2.6.12-rc4-love3/scripts/reference_init.pl	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/scripts/reference_init.pl	2006-04-26 19:09:36.753759552 +1200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+#!/System/Tools/perl -w
 #
 # reference_init.pl (C) Keith Owens 2002 <kaos@ocs.com.au>
 #
diff -urN linux-2.6.12-rc4-love3/scripts/show_delta linux-2.6.12-rc4-love3-STEP/scripts/show_delta
--- linux-2.6.12-rc4-love3/scripts/show_delta	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/scripts/show_delta	2006-04-26 19:09:36.754759400 +1200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/System/Tools/env python
 #
 # show_deltas: Read list of printk messages instrumented with
 # time data, and format with time deltas.
diff -urN linux-2.6.12-rc4-love3/scripts/ver_linux linux-2.6.12-rc4-love3-STEP/scripts/ver_linux
--- linux-2.6.12-rc4-love3/scripts/ver_linux	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/scripts/ver_linux	2006-04-26 19:09:36.756759096 +1200
@@ -1,10 +1,10 @@
-#!/bin/sh
+#!/System/Tools/sh
 # Before running this script please ensure that your PATH is
 # typical as you use for compilation/istallation. I use
 # /bin /sbin /usr/bin /usr/sbin /usr/local/bin, but it may
 # differ on your system.
 #
-PATH=/sbin:/usr/sbin:/bin:/usr/bin:/usr/local/sbin:$PATH
+PATH=/System/Tools:/System/Library/Tools:$PATH
 echo 'If some fields are empty or look unusual you may have an old version.'
 echo 'Compare to the current minimal requirements in Documentation/Changes.'
 echo ' '
@@ -61,14 +61,14 @@
 showmount --version 2>&1 | grep nfs-utils | awk \
 'NR==1{print "nfs-utils             ", $NF}'
 
-ls -l `ldd /bin/sh | awk '/libc/{print $3}'` | sed \
+ls -l `ldd /System/Tools/sh | awk '/libc/{print $3}'` | sed \
 -e 's/\.so$//' | awk -F'[.-]'   '{print "Linux C Library        " \
 $(NF-2)"."$(NF-1)"."$NF}'
 
-ldd -v > /dev/null 2>&1 && ldd -v || ldd --version |head -n 1 | awk \
+ldd -v > /System/Devices/null 2>&1 && ldd -v || ldd --version |head -n 1 | awk \
 'NR==1{print "Dynamic linker (ldd)  ", $NF}'
 
-ls -l /usr/lib/lib{g,stdc}++.so  2>/dev/null | awk -F. \
+ls -l /usr/lib/lib{g,stdc}++.so  2>/System/Devices/null | awk -F. \
        '{print "Linux C++ Library      " $4"."$5"."$6}'
 
 ps --version 2>&1 | grep version | awk \
diff -urN linux-2.6.12-rc4-love3/security/keys/proc.c linux-2.6.12-rc4-love3-STEP/security/keys/proc.c
--- linux-2.6.12-rc4-love3/security/keys/proc.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/security/keys/proc.c	2006-04-26 19:09:36.757758944 +1200
@@ -72,14 +72,14 @@
 #ifdef CONFIG_KEYS_DEBUG_PROC_KEYS
 	p = create_proc_entry("keys", 0, NULL);
 	if (!p)
-		panic("Cannot create /proc/keys\n");
+		panic("Cannot create /System/Processes/keys\n");
 
 	p->proc_fops = &proc_keys_fops;
 #endif
 
 	p = create_proc_entry("key-users", 0, NULL);
 	if (!p)
-		panic("Cannot create /proc/key-users\n");
+		panic("Cannot create /System/Processes/key-users\n");
 
 	p->proc_fops = &proc_key_users_fops;
 
diff -urN linux-2.6.12-rc4-love3/security/keys/request_key.c linux-2.6.12-rc4-love3-STEP/security/keys/request_key.c
--- linux-2.6.12-rc4-love3/security/keys/request_key.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/security/keys/request_key.c	2006-04-26 19:09:36.758758792 +1200
@@ -71,12 +71,12 @@
 	/* set up a minimal environment */
 	i = 0;
 	envp[i++] = "HOME=/";
-	envp[i++] = "PATH=/sbin:/bin:/usr/sbin:/usr/bin";
+	envp[i++] = "PATH=/System/Tools:/System/Library/Tools";
 	envp[i] = NULL;
 
 	/* set up the argument list */
 	i = 0;
-	argv[i++] = "/sbin/request-key";
+	argv[i++] = "/System/Tools/request-key";
 	argv[i++] = (char *) op;
 	argv[i++] = key_str;
 	argv[i++] = uid_str;
diff -urN linux-2.6.12-rc4-love3/sound/drivers/dummy.c linux-2.6.12-rc4-love3-STEP/sound/drivers/dummy.c
--- linux-2.6.12-rc4-love3/sound/drivers/dummy.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/sound/drivers/dummy.c	2006-04-26 19:09:36.760758488 +1200
@@ -32,7 +32,7 @@
 #include <sound/initval.h>
 
 MODULE_AUTHOR("Jaroslav Kysela <perex@suse.cz>");
-MODULE_DESCRIPTION("Dummy soundcard (/dev/null)");
+MODULE_DESCRIPTION("Dummy soundcard (/System/Devices/null)");
 MODULE_LICENSE("GPL");
 MODULE_SUPPORTED_DEVICE("{{ALSA,Dummy soundcard}}");
 
diff -urN linux-2.6.12-rc4-love3/sound/oss/emu10k1/main.c linux-2.6.12-rc4-love3-STEP/sound/oss/emu10k1/main.c
--- linux-2.6.12-rc4-love3/sound/oss/emu10k1/main.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/sound/oss/emu10k1/main.c	2006-04-26 19:09:36.762758184 +1200
@@ -211,7 +211,7 @@
 
 		card->waveout.send_routing[ROUTE_PCM] = 0x3210; // pcm
 		card->waveout.send_routing[ROUTE_PT] = 0x3210; // passthrough
-		card->waveout.send_routing[ROUTE_PCM1] = 0x7654; // /dev/dsp1
+		card->waveout.send_routing[ROUTE_PCM1] = 0x7654; // /System/Devices/dsp1
 	}
 
 	/* Assign default recording parameters */
@@ -325,14 +325,14 @@
 	len += sprintf (page + len, "IO             : %#06lx-%#06lx\n", card->iobase, card->iobase + card->length - 1);
 	len += sprintf (page + len, "IRQ            : %d\n\n", card->irq);
 	
-	len += sprintf (page + len, "Registered /dev Entries:\n");
-	len += sprintf (page + len, "/dev/dsp%d\n", card->audio_dev / 16);
-	len += sprintf (page + len, "/dev/dsp%d\n", card->audio_dev1 / 16);
-	len += sprintf (page + len, "/dev/mixer%d\n", card->ac97->dev_mixer / 16);
-	len += sprintf (page + len, "/dev/midi%d\n", card->midi_dev / 16);
+	len += sprintf (page + len, "Registered /System/Devices Entries:\n");
+	len += sprintf (page + len, "/System/Devices/dsp%d\n", card->audio_dev / 16);
+	len += sprintf (page + len, "/System/Devices/dsp%d\n", card->audio_dev1 / 16);
+	len += sprintf (page + len, "/System/Devices/mixer%d\n", card->ac97->dev_mixer / 16);
+	len += sprintf (page + len, "/System/Devices/midi%d\n", card->midi_dev / 16);
 
 #ifdef EMU10K1_SEQUENCER
-	len += sprintf (page + len, "/dev/sequencer\n");
+	len += sprintf (page + len, "/System/Devices/sequencer\n");
 #endif
 
 	return len;
diff -urN linux-2.6.12-rc4-love3/sound/oss/forte.c linux-2.6.12-rc4-love3-STEP/sound/oss/forte.c
--- linux-2.6.12-rc4-love3/sound/oss/forte.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/sound/oss/forte.c	2006-04-26 19:09:36.764757880 +1200
@@ -1966,7 +1966,7 @@
 
 	/* Register with /proc */
 	if (forte_proc_init()) {
-		printk (KERN_ERR PFX "couldn't add entries to /proc!\n");
+		printk (KERN_ERR PFX "couldn't add entries to /System/Processes!\n");
 		return -1;
 	}
 
diff -urN linux-2.6.12-rc4-love3/sound/usb/usx2y/usX2Yhwdep.c linux-2.6.12-rc4-love3-STEP/sound/usb/usx2y/usX2Yhwdep.c
--- linux-2.6.12-rc4-love3/sound/usb/usx2y/usX2Yhwdep.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/sound/usb/usx2y/usX2Yhwdep.c	2006-04-26 19:09:36.765757728 +1200
@@ -274,7 +274,7 @@
 	hw->ops.mmap = snd_us428ctls_mmap;
 	hw->ops.poll = snd_us428ctls_poll;
 	hw->exclusive = 1;
-	sprintf(hw->name, "/proc/bus/usb/%03d/%03d", device->bus->busnum, device->devnum);
+	sprintf(hw->name, "/System/Processes/bus/usb/%03d/%03d", device->bus->busnum, device->devnum);
 	return 0;
 }
 
diff -urN linux-2.6.12-rc4-love3/sound/usb/usx2y/usx2yhwdeppcm.c linux-2.6.12-rc4-love3-STEP/sound/usb/usx2y/usx2yhwdeppcm.c
--- linux-2.6.12-rc4-love3/sound/usb/usx2y/usx2yhwdeppcm.c	2005-05-07 17:20:31.000000000 +1200
+++ linux-2.6.12-rc4-love3-STEP/sound/usb/usx2y/usx2yhwdeppcm.c	2006-04-26 19:09:36.767757424 +1200
@@ -767,7 +767,7 @@
 	hw->ops.release = snd_usX2Y_hwdep_pcm_release;
 	hw->ops.mmap = snd_usX2Y_hwdep_pcm_mmap;
 	hw->exclusive = 1;
-	sprintf(hw->name, "/proc/bus/usb/%03d/%03d/hwdeppcm", dev->bus->busnum, dev->devnum);
+	sprintf(hw->name, "/System/Processes/bus/usb/%03d/%03d/hwdeppcm", dev->bus->busnum, dev->devnum);
 
 	err = snd_pcm_new(card, NAME_ALLCAPS" hwdep Audio", 2, 1, 1, &pcm);
 	if (err < 0) {
