diff -urN shadow-4.0.3/libmisc/login_access.c shadow-4.0.3-uclibc/libmisc/login_access.c
--- shadow-4.0.3/libmisc/login_access.c	1998-01-30 12:22:34.000000000 +1300
+++ shadow-4.0.3-uclibc/libmisc/login_access.c	2008-04-20 20:16:50.978413652 +1200
@@ -205,6 +205,7 @@
     return (NO);
 #endif
 #else  /* works better with glibc? */
+#ifdef NIS
 	static char *mydomain = 0;
 
 	if (mydomain == 0) {
@@ -215,6 +216,10 @@
 	}
 
 	return innetgr(group, machine, user, mydomain);
+#else
+    syslog(LOG_ERR, "NIS netgroup support not configured");              
+    return (NO);                                                  
+#endif
 #endif
 }
 
diff -urN shadow-4.0.3/libmisc/xmalloc.c shadow-4.0.3-uclibc/libmisc/xmalloc.c
--- shadow-4.0.3/libmisc/xmalloc.c	1998-12-29 09:34:56.000000000 +1300
+++ shadow-4.0.3-uclibc/libmisc/xmalloc.c	2008-04-20 20:17:03.250406140 +1200
@@ -16,7 +16,7 @@
 
 #include "defines.h"
 
-extern char *malloc();
+extern char *xmalloc();
 
 char *
 xmalloc(size_t size)
